src/mmw/js/src/compare/controllers.js

```diff
@@ -16,28 +18,87 @@ var CompareController = {
             App.currProject
                 .fetch()
                 .done(function() {
+                    setupProjectCopy();
```

Comment:
If the user is a guest, goes into the compare views, and then logs in, the project should be saved with the inputs and results that were present before going to the compare views. This is to enforce the constraint that inputs and results entered in the compare views should never be saved. Without holding onto the original copies of these values, it's not possible to do this.